### ISSUE TITLE
Creating a project from a URL: URLs with leading whitespace are rejected

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#2516]
**Describe the bug**

The input field to create a project from a URL does not trim its input when validating it.

**To Reproduce**
1. Go to 'Create Project' -> 'Web Addresses (URL)'
2. Paste "  http://api.worldbank.org/countries/all/indicators/SP.POP.TOTL?date=2000:2001" with the leading whitespace
3. Validate
4. The URL is rejected

**Expected behavior**

The value of this input should be trimmed before validation.

**OpenRefine (please complete the following information):**
 - master branch (3.4 snapshot)

**Comment by Nishtha3512** [comment]
![Screenshot from 2020-04-04 14-36-17](https://user-images.githubusercontent.com/51858166/78423107-c8cda300-7681-11ea-9517-37e59309b3ac.png)
Hey,Where is the option to validate?
As you can see in above screenshot when i click on create project then project is simply created without rejecting any URL.

**Comment by wetneb** [comment]
Are you sure you included whitespace before the URL?
![image](https://user-images.githubusercontent.com/309908/78423384-408dd480-7666-11ea-8481-4e99a9f57284.png)

**Comment by Nishtha3512** [comment]
Ohh now i understand your issue.Can you please tell me where  is code related to issue?I will try to fix it. 
Is it in OpenRefine/main/src/com/google/refine/importers/XmlImporter.java file?

**Comment by wetneb** [comment]
No, it is a UI issue, so it is in a Javascript file.

Here is how you can find it:
 * Search for the error message "You must specify a valid web address" in the source tree: `grep -R "You must specify a valid web address" .`
 * You will find this message in the transations: `"core-index-import/warning-web-address": "You must specify a valid web address (URL) to import."`
* Now search for the identifier of the message: `grep -R "core-index-import/warning-web-address" .`.
* The only Javascript file that contains this is `./main/webapp/modules/core/scripts/index/default-importing-sources/sources.js`, so the error is coming from there.
* Open this file and look for `warning-web-address`.

With this method you do not need to understand OpenRefine's structure much: you can locate the code you are interested in just by following references, starting from UI text.

**Comment by kushthedude** [comment]
I think we can just add a function in UI to trim the whitespaces in the input URL before it is sent to the Regex for validation, or can we just add a trim function in the regex function?

**Comment by Nishtha3512** [comment]
> I think we can just add a function in UI to trim the whitespaces in the input URL before it is sent to the Regex for validation, or can we just add a trim function in the regex function?

Yes am also trying to add trim function to resolve this issue :)
